libensemble/resources.py

```diff
@@ -150,13 +150,23 @@ def get_MPI_variant():
         except OSError:
             pass
 
-        # Explore mpi4py.MPI.get_vendor() and mpi4py.MPI.Get_library_version() for mpi4py
-        try_mpich = subprocess.Popen(['mpirun', '-npernode'], stdout=subprocess.PIPE,
-                                     stderr=subprocess.STDOUT)
-        stdout, _ = try_mpich.communicate()
-        if 'unrecognized argument npernode' in stdout.decode():
-            return 'mpich'
-        return 'openmpi'
+        try:
+            # Explore mpi4py.MPI.get_vendor() and mpi4py.MPI.Get_library_version() for mpi4py
+            try_mpich = subprocess.Popen(['mpirun', '-npernode'], stdout=subprocess.PIPE,
+                                         stderr=subprocess.STDOUT)
+            stdout, _ = try_mpich.communicate()
+            if 'unrecognized argument npernode' in stdout.decode():
+                return 'mpich'
+            return 'openmpi'
```

Comment:
Steve,
Are we assuming only mpich or openmpi? (Seems safe. Should we make it explicit somewhere?)

Reply:
The process looks for other launch commands like aprun/jsrun. If it comes down to mpirun, this is testing whether its using mpich or openMPI as a base, which have different options to mpirun. So yes, currently mpirun is assumed to be one of those. Intel MPI, for example, is using the mpich commands. I'm not sure what other implementations there may be.